test/integration/freetext_editor_spec.js

```diff
@@ -252,4 +252,176 @@ describe("Editor", () => {
+            await page.mouse.click(
+              editorRect.x,
+              editorRect.y + 2 * editorRect.height
+            );
+          }
+
+          await page.keyboard.down("Control");
+          await page.keyboard.press("a");
+          await page.keyboard.up("Control");
+
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([0, 1, 2, 3]);
+
+          await page.keyboard.down("Control");
+          await page.mouse.click(editorCenters[1].x, editorCenters[1].y);
+
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([0, 2, 3]);
+
+          await page.mouse.click(editorCenters[2].x, editorCenters[2].y);
+
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([0, 3]);
+
+          await page.mouse.click(editorCenters[1].x, editorCenters[1].y);
+          await page.keyboard.up("Control");
+
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([0, 1, 3]);
+
+          await page.keyboard.down("Control");
+          await page.keyboard.press("c");
+          await page.keyboard.up("Control");
+
+          await page.keyboard.down("Control");
+          await page.keyboard.press("v");
+          await page.keyboard.up("Control");
+
+          // 0,1,3 are unselected and new paster editors are selected.
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([4, 5, 6]);
+
+          // No ctrl here, hence all are unselected and 2 is selected.
+          await page.mouse.click(editorCenters[2].x, editorCenters[2].y);
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([2]);
+
+          await page.mouse.click(editorCenters[1].x, editorCenters[1].y);
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([1]);
+
+          await page.keyboard.down("Control");
+
+          await page.mouse.click(editorCenters[3].x, editorCenters[3].y);
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([1, 3]);
+
+          await page.keyboard.up("Control");
+
+          // Delete 1 and 3.
+          await page.keyboard.press("Backspace");
+
+          await page.keyboard.down("Control");
+          await page.keyboard.press("a");
+          await page.keyboard.up("Control");
+
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([0, 2, 4, 5, 6]);
+
+          // Create an empty editor.
+          await page.mouse.click(rect.x + 700, rect.y + 100);
+          expect(await getSelected(page))
+            .withContext(`In ${browserName}`)
+            .toEqual([7]);
+
+          // Get the focus to 2 and check that only 2 is selected.
```

Comment:
Nit: Did you mean the following?
```suggestion
          // Set the focus to 2 and check that only 2 is selected.
```